### DISCUSSION TITLE
Intention of `maximumAmountOfResults` for data lake measurement API

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#1318] [Dev] [answered]
What's the intention behind the `maximumAmountOfResults` query parameter for data lake measurements?
There is already the parameter `limit` which allows me to restrict the number of returned results.
Contrastively, `maximumAmountOfResults` simply returns `TOO_MUCH_DATA` when there are more data records then the specified amount.

**Comment by tenthe** [comment, marked as answer]
Hi @bossenti,
this is a good question and in fact this is not described very clearly.

The limit can be set by the user to retrieve only a certain amount of data:
![grafik](https://user-images.githubusercontent.com/5279561/220680842-3057c401-4b48-4193-b4a9-79da913eb08e.png)

With the parameter `maximumAmountOfResults` you can make sure that not too much data is loaded into the browser.
Should this be the case, the user will receive a warning message.

**Reply by bossenti** [reply]
But where is `maximumAmountOfResults` set? 
It isn't done by the user, is it?

**Reply by tenthe** [reply]
This is set by the client. (e.g. for data explorer widgets in `BaseDataExplorerWidgetDirective.TOO_MUCH_DATA_PARAMETER`)
The main idea was that a client can decide how much data it can handle.
